src/averageAges.js

```diff
@@ -37,7 +50,28 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people.filter(person => person.sex === 'f');
+  const averageAgeWomen = women
+    .reduce(
+      (sum, person) => sum + (person.died - person.born) / women.length,
+      0,
+    );
+
+  const child = people
+    .filter(
+      girl => people.some(person => person.name === girl.mother)
+    );
```

Comment:
```suggestion
  const child = people.filter(
      girl => people.some(person => person.name === girl.mother)
    );
```
ты можешь оставить так как у тебя, но вообще пишут так